packages/aws-cdk/lib/context-providers/vpcs.ts

```diff
@@ -0,0 +1,193 @@
+import cxapi = require('@aws-cdk/cx-api');
+import AWS = require('aws-sdk');
+import { Mode, SDK } from '../api';
+import { debug } from '../logging';
+import { ContextProviderPlugin } from './provider';
+
+export class VpcNetworkContextProviderPlugin implements ContextProviderPlugin {
+
+  constructor(private readonly aws: SDK) {
+  }
+
+  public async getValue(args: cxapi.VpcContextQuery) {
+    const account: string = args.account;
+    const region: string = args.region;
+
+    const ec2 = await this.aws.ec2(account, region, Mode.ForReading);
+
+    const vpcId = await this.findVpc(ec2, args);
+
+    return await this.readVpcProps(ec2, vpcId);
+  }
+
+  private async findVpc(ec2: AWS.EC2, args: {[key: string]: any}): Promise<string> {
+    const vpcId: string | undefined = args.vpcId;
+    const vpcName: string | undefined = args.vpcName;
+    const tags: {[key: string]: string} | undefined = args.tags;
+    const isDefault: boolean | undefined = args.isDefault;
+
+    // Builter request filter
+    const filters: AWS.EC2.Filter[] = [];
+    if (vpcId) { filters.push({ Name: 'vpc-id', Values: [vpcId] }); }
+    if (vpcName) { filters.push({ Name: 'tag:Name', Values: [vpcName] }); }
+    if (tags) {
+      for (const [tag, value] of Object.entries(tags)) {
+        filters.push({ Name: `tag:${tag}`, Values: [value] });
+      }
+    }
+    if (isDefault !== undefined) {
+      filters.push({ Name: 'isDefault', Values: [isDefault ? 'true' : 'false'] });
+    }
+
+    debug(`Listing VPCs in ${args.account}:${args.region}`);
+    const response = await ec2.describeVpcs({ Filters: filters }).promise();
+
+    const vpcs = response.Vpcs || [];
+    if (vpcs.length === 0) {
+      throw new Error(`Could not find any VPCs matching ${JSON.stringify(args)}`);
+    }
+    if (vpcs.length > 1) {
+      throw new Error(`Found ${vpcs.length} VPCs matching ${JSON.stringify(args)}; please narrow the search criteria`);
+    }
+
+    return vpcs[0].VpcId!;
+  }
+
+  private async readVpcProps(ec2: AWS.EC2, vpcId: string): Promise<cxapi.VpcContextResponse> {
+    debug(`Describing VPC ${vpcId}`);
+
+    const response = await ec2.describeSubnets({ Filters: [{ Name: 'vpc-id', Values: [vpcId] }] }).promise();
+    const listedSubnets = response.Subnets || [];
+
+    // Now comes our job to separate these subnets out into AZs and subnet groups (Public, Private, Isolated)
+    // We have the following attributes to go on:
+    // - Type tag, we tag subnets with their type
+    // - Name tag, we tag subnets with their subnet group name
+    // - MapPublicIpOnLaunch in absence of tags => must be a Public subnet, anything else is either Isolated or Private
+
+    const azs = Array.from(new Set<string>(listedSubnets.map(s => s.AvailabilityZone!)));
+    azs.sort();
+
+    const subnets: Subnet[] = listedSubnets.map(subnet => {
+      let type = getTag('aws-cdk:SubnetType', subnet.Tags);
```

Comment:
I think we try to standardize around something like `aws:cdk:subnet-type`

Reply:
can we set an `aws:` tag? or does it have to be `aws-cdk`?

Reply:
Don't believe we can set the `aws:` tag.